src/System.Private.CoreLib/src/System/Enum.CoreCLR.cs

```diff
@@ -109,5 +145,36 @@ private static RuntimeType ValidateRuntimeType(Type enumType)
                 throw new ArgumentException(SR.Arg_MustBeType, nameof(enumType));
             return rtType;
         }
+
+        public int CompareTo(object target)
+        {
+            const int retIncompatibleMethodTables = 2;  // indicates that the method tables did not match
+            const int retInvalidEnumType = 3; // indicates that the enum was of an unknown/unsupported underlying type
+
+            if (this == null)
```

Comment:
Just curious, why is this needed?

Reply:
Explained here: https://github.com/dotnet/coreclr/pull/22564#discussion_r256241935

The FCall expects that the `this` is non-null.